### ISSUE TITLE
Possibility to display more data other than "name" 

### ISSUE THREAD

**Issue by sikandarchishty** [jpb12/react-tree-graph#80]
Hi,

Is it possible to Display other data as well? string/image

or a reusable component as a whole instead of "name".

Concept:
![Concept](https://user-images.githubusercontent.com/19312954/152887542-7ec6c624-1f87-425a-828d-17ae69a5b17d.png)

BR,
Sikandar

**Comment by jpb12** [comment]
https://github.com/jpb12/react-tree-graph#configuration

You can change `labelProp` to something other than name to render that instead. This property can be a react node instead of a string.

```javascript
let data = {
	name: 'Parent',
	children: [...]
        label: <MyCustomComponent .../>
};

<Tree
	data={data}
	height={400}
	width={400}
        labelProp="label"/>);
```

**Comment by jpb12** [comment]
This was a combination of two issues:
* The clone library we were using did not support cloning react components. It seems d3 updated at some point to no longer modify the object passed in, so cloning is no longer necessary and has been removed.
* The label was always rendered inside a `<text>` node, which prevented anything other than text from being rendered.

I've fixed both issues and released a new version, 7.0.1. There are a few restrictions:
* If you are not rendering a string, you must render valid svg elements (no divs).
* If you are rendering a custom react node, you must wrap any text inside a `<text>` node.
* You can return multiple nodes within a react fragment, as everything will be wrapped inside a `<g>` node.

If there are any issues, please let me know and reopen this ticket.

**Comment by jpb12** [comment]
I've updated the documentation with an [example](https://jpb12.github.io/react-tree-graph/#custom-labels).
